helm/tests/ensure.py

```diff
@@ -773,3 +773,96 @@ def kubeadm_control_plane(kubernetes_cluster):
 
     kubernetes_cluster.kubectl(f"delete kubeadmcontrolplane {cluster_name}", output=None)
     LOGGER.info(f"kubeadmcontrolplane {cluster_name} deleted")
+
+# Kyverno fixtures
+
+@pytest.fixture
+def fetch_policies(kubernetes_cluster):
+    raw = kubernetes_cluster.kubectl(
+        f"get cpol", output="yaml")
+
+    kcp = yaml.safe_load(raw)
+
+    yield kcp
+
+@pytest.fixture
+def run_pod_from_registries(kubernetes_cluster):
+    bad_pod_name = "pod-outside-gs-registries"
+    good_pod_name = "pod-inside-gs-registries"
+
+    # Create a Pod with a known bad registry
+    c = dedent(f"""
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          labels:
+            cluster.x-k8s.io/cluster-name: {cluster_name}
+          name: {bad_pod_name}
+          namespace: default
+        spec:
+          containers:
+          - name: bad-registry
+            image: docker.io/nginx
+    """)
+
+    kubernetes_cluster.kubectl("apply", input=c, output=None)
+    LOGGER.info(f"Pod {bad_pod_name} applied")
+
+    # Create a Pod with all the known good registries
+    c = dedent(f"""
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          labels:
+            cluster.x-k8s.io/cluster-name: {cluster_name}
+          name: {good_pod_name}
+          namespace: default
+        spec:
+          containers:
+          - name: good-registry-quay
+            image: quay.io/giantswarm/nginx
+          - name: good-registry-docker
+            image: docker.io/giantswarm/nginx
+          - name: good-registry-gs
+            image: giantswarm/nginx
+          - name: good-registry-azurecr
+            image: giantswarm.azurecr.io/giantswarm/nginx
+          - name: good-registry-aliyuncs
+            image: registry-intl.cn-shanghai.aliyuncs.com/giantswarm/nginx
+    """)
+
+    kubernetes_cluster.kubectl("apply", input=c, output=None)
+    LOGGER.info(f"Pod {good_pod_name} applied")
+
+    # Wait for a PolicyReport to be created
+    timeout = 0
+    reports_found = False
+
+    while timeout < 5 and not reports_found:    
+      polr = yaml.safe_load(kubernetes_cluster.kubectl(
+          f"get polr", output="yaml"))
+
+      if len(polr['items']) == 0:
+        time.sleep(15)
+        timeout += 1
+      else:
+        # Check that there is one result for each pod
+        for report in polr['items']:
```

Comment:
I feel like these could be combined, since if the length is 0 you just won't have any reports. You might not need the `len(polr['items'])` check. Not a big deal though

Reply:
Here the problem is that as there are two pods, we need to have both reports.

First check looks for `polr-ns-default` and the second one ensures that both pods are already inside that report. I had a couple of runs where the second pod was not part of the report yet, causing it to fail.